core/Version.php

```diff
@@ -56,4 +56,28 @@ public function isPreviewVersion($version): bool
 
         return false;
     }
+
+    public function nextPreviewVersion($version): string
+    {
+        if (!$this->isVersionNumber($version)) {
+            return '';
+        }
+
+        $dt = date('YmdHis');
+
+        if ($this->isPreviewVersion($version)) {
+            // already a preview, update dt and check it's newer
+            $newVersion = substr($version, 0, -14) . $dt;
+            if (version_compare($version, $newVersion, '<')) {
+                return $newVersion;
+            }
+            return '';
+        } elseif ($this->isStableVersion($version)) {
+            // no suffix yet
+            return $version . '-alpha.' . $dt;
+        } else {
+            // -b1, -rc1, -alpha
+            return $version . '.' . $dt;
```

Comment:
Before simply appending that to an unstable release (other than alpha), I guess we would need to check if that version has already been released.

Reply:
That wouldn't be a concern for this method though. This provides the next preview based on what's in core/Version.php, in principle. Is there a possibility we will bump the version in core/Version.php and not release a version at the same time? I guess it can happen, but should it happen and would it be expected to happen at 1am NZ time? Happy to adjust if we deem it necessary, just wondering how likely it is to happen.

Reply:
But if we assume that the version stated in this file was already release, we would need to increase the beta / rc number as well, to ensure the returned version is higher.

Reply:
I'll add a check to see if the version in the version file has been released already in case of stable and non-stable versions (beta, rc). If there's no such version this workflow will stop.

Reply:
This has been added and can be seen at https://github.com/matomo-org/matomo/pull/22367/files#diff-a8159e7874b5e3a5989b163a4ab774a38cdf4060985ecfc473e3f784479e5612R99